File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        int n=nums.length;
        HashMap<Integer,Integer>mp=new HashMap<>();
        for(int x:nums){
            mp.put(x,mp.getOrDefault(x,0)+1);
        }    
        for(int i=k;;i+=k){
            if(!mp.containsKey(i)){
                return i;
            }
        }
      
    }
}

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        int n=nums.size();
        int ans=0;
        unordered_map<int,int>mp;
        for(auto it:nums){
            if(it%k==0){
                mp[it]++;
            }
        }
        for(int i=k;;i+=k){
            if(mp.find(i)==mp.end()) return i;
        }
        return -1;

    }
};